src/browser.ts

```diff
@@ -0,0 +1,212 @@
+ * notice, this list of conditions and the following disclaimer.
+ * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import * as protobuf from 'protobufjs';
+import * as gax from './gax';
+import {Status} from './status';
+import {OutgoingHttpHeaders} from 'http';
+import {GoogleAuth} from 'google-auth-library';
+import {OperationsClientBuilder} from './operationsClientBrowser';
+import {GrpcClientOptions, ClientStubOptions} from './grpc';
+import {GaxCall, GRPCCall} from './apitypes';
+import {Descriptor} from './descriptor';
+import {createApiCall as _createApiCall} from './createApiCall';
+export {CallSettings, constructSettings, RetryOptions} from './gax';
+
+export {
+  BundleDescriptor,
+  LongrunningDescriptor,
+  PageDescriptor,
+  StreamDescriptor,
+} from './descriptor';
+
+export {StreamType} from './streamingCalls/streaming';
+
+export const ALL_SCOPES: string[] = [];
+lro.ALL_SCOPES = ALL_SCOPES;
+
+interface CancelHandler {
+  canceller: AbortController;
+  cancelRequested: boolean;
+}
+
+export class GrpcClient {
+  auth: GoogleAuth;
+  promise?: PromiseConstructor;
+
+  constructor(options: GrpcClientOptions = {}) {
+    if (!options.auth) {
+      throw new Error(
+        'You need to pass auth instance to gRPC-fallback client. Use OAuth2Client from google-auth-library.'
+      );
+    }
+    this.auth = options.auth;
+    this.promise = 'promise' in options ? options.promise! : Promise;
+  }
+
+  loadProto(jsonObject) {
+    const rootObject = protobuf.Root.fromJSON(jsonObject);
+    return rootObject;
+  }
+
+  private getServiceMethods(service: protobuf.Service) {
+    const methods = Object.keys(service.methods);
+
+    // Creating an array of all methods within the stub with correct case (i.e, first letter is lower-case)
```

Comment:
Comment here is maybe unnecessary

Reply:
Unnecessary indeed since variable name speaks for itself.